v2/datastream-to-spanner/src/main/java/com/google/cloud/teleport/v2/templates/SpannerTransactionWriterDoFn.java

```diff
@@ -304,6 +313,21 @@ JsonNode convertTableAndColumnNames(JsonNode changeEvent, String tableName) {
     return changeEvent;
   }
 
+  JsonNode populateShardId(JsonNode changeEvent, String tableId) {
+    if (shardingConfig.isEmpty()) {
+      return changeEvent; // Nothing to do
+    }
+    CreateTable table = session.getSpSchema().get(tableId);
+    ColumnDef shardedColDef = table.getColDefs().get(table.getShardedColId());
```

Comment:
How does the getShardedColId() behave for non-sharded migrations? Will it always be present in the session file?